recipes-soletta/dev-app/soletta-dev-app_0.1.bb

```diff
@@ -31,6 +35,60 @@ FILES_${PN} += " \
     /soletta-dev-app/scripts/soletta-dev-app-mac.sh \
 "
 
+do_compile() {
+
+    # changing the home directory to the working directory, the .npmrc will be created in this directory
+    export HOME=${WORKDIR}
+
+    # does not build dev packages
+    npm config set dev false
+
+    # access npm registry using http
+    npm set strict-ssl false
+    npm config set registry http://registry.npmjs.org/
+
+    # configure http proxy if neccessary
+    if [ -n "${http_proxy}" ]; then
+        npm config set proxy ${http_proxy}
+        export bower_https_proxy=${http_proxy}
+    fi
+    if [ -n "${HTTP_PROXY}" ]; then
+        npm config set proxy ${HTTP_PROXY}
+        export bower_https_proxy=${HTTP_PROXY}
+    fi
+
+    # configure cache to be in working directory
+    npm set cache ${WORKDIR}/npm_cache
+
+    # clear local cache prior to each compile
+    npm cache clear
+
+    case ${TARGET_ARCH} in
+        i?86) targetArch="ia32"
+            echo "targetArch = 32"
+            ;;
+        x86_64) targetArch="x64"
+            echo "targetArch = 64"
+            ;;
+        arm) targetArch="arm"
+            ;;
+        mips) targetArch="mips"
+            ;;
+        sparc) targetArch="sparc"
+            ;;
+        *) echo "unknown architecture"
+           exit 1
+            ;;
+    esac
+
+    npm install --arch=${targetArch} --production --verbose -g bower@v1.6.8
```

Comment:
I wonder if we could just package the `npm` dependencies in this layer and skip calling npm altogether. Such as we pack `bower` and depend on that package.